app/models/case/base.rb

```diff
@@ -342,6 +344,7 @@ def create!(_attributes = {})
 
   after_initialize do
     self.workflow = default_workflow if self.workflow.nil?
+    @deadline_calculator = self.class.respond_to?(:type_abbreviation) ? create_deadline_calculator : nil
```

Comment:
Not sure what this is doing - may be good to wrap it in a descriptive method possibly.